src/lib/openjp2/mqc.c

```diff
@@ -203,7 +203,7 @@ static opj_mqc_state_t mqc_states[47 * 2] = {
 */
 
 static void opj_mqc_byteout(opj_mqc_t *mqc) {
-	if (*mqc->bp == 0xff) {
+	if ( (mqc->bp >= mqc->start) & (*mqc->bp == 0xff)) {
```

Comment:
No, & is superior because it avoids a branch in the code - C has a short circuit on logical operators so there would be two branches with &&